Modules/_elementtree.c

```diff
@@ -3411,7 +3411,10 @@ xmlparser_gc_traverse(XMLParserObject *self, visitproc visit, void *arg)
 static int
 xmlparser_gc_clear(XMLParserObject *self)
 {
-    EXPAT(ParserFree)(self->parser);
+    if (self->parser != NULL) {
+        EXPAT(ParserFree)(self->parser);
```

Comment:
Perhaps it would be safer to set `self->parser = NULL` before calling `ParserFree`.

Reply:
IMHO it's overkill. EXPAT(ParserFree) calls a C function which is outside the "Python" world, I don't see how it could trigger a reentrant call.

Reply:
Well, I made the change anyway. It doesn't harm much :-)